New Assumptions: Make list of failing cases involving binary predicate

TiloRC opened the pull request:
SymPy can’t do much with relational assumptions at the moment. This is a list of things that it can’t do that I would like to implement as part of GSoC.

<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234" (see
https://tinyurl.com/auto-closing for more information). Also, please
write a comment on that issue linking back to this pull request once it is
open. -->






<!-- Write the release notes for this release below between the BEGIN and END
statements. The basic format is a bulleted list with the name of the subpackage
and the release note for this PR. For example:

* solvers
  * Added a new solver for logarithmic equations.

* functions
  * Fixed a bug with log of integers. Formerly, `log(-x)` incorrectly gave `-log(x)`.

* physics.units
  * Corrected a semantical error in the conversion between volt and statvolt which
    reported the volt as being larger than the statvolt.

or if no release note(s) should be included use:

NO ENTRY

See https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more
information on how to write release notes. The bot will check your release
notes automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->

<!-- END RELEASE NOTES -->

TiloRC committed:
New Assumptions: Make list of failing cases involving binary predicate

SymPy can’t do much with relational assumptions at the moment. This is a
list of things that it can’t do that I would like to implement as part of GSoC.

asmeurer commented on sympy/assumptions/tests/test_failing_relational_queries.py:
```diff
@@ -0,0 +1,37 @@
+    assert ask(Implies(And(Q.lt(x, y), Q.negative(y)), Q.negative(x))) is True
+
+    # test more complex problems
+    assert ask(x > z, x > y & y > z)
```
& has a higher precedence than > so you have to parenthesize things like this. 

```suggestion
    assert ask(x > z, (x > y) & (y > z))
```

TiloRC committed:
Add failing cases for relational, refine, and piecewise

TiloRC committed:
Merge branch 'master' into relational-fails

TiloRC committed:
Merge master into branch for checks and fix import

asmeurer commented on sympy/assumptions/tests/test_failing_relational_queries.py:
```diff
@@ -0,0 +1,69 @@
+@XFAIL
+def test_relational():
+    with assuming(x > 1):
+        assert x > 1 == True
```
```suggestion
        assert (x > 1) == True
```

asmeurer replied:
Actually this should probably be using `refine(x > 1)` or `ask(x > 1)`

TiloRC committed:
Add number line prop tests and improve old tests

TiloRC committed:
Remove unicode

TiloRC committed:
use == instead of is in test piecewise

TiloRC committed:
Update test_failing_relational_queries.py

Added transitive property to test equality and removed converse property from test number line properties

TiloRC committed:
Update test_failing_relational_queries.py

x > y implies x and y are real numbers because inequalities don't make sense of complex numbers with an imaginary component

TiloRC committed:
inequalities imply variables are extended_real, not real

x could be infinite if x > 0

TiloRC committed:
Rename test_failing_relational_queries.py to failing_queries.py

TiloRC committed:
Update failing_queries.py

Added missing @XFAIL
